### PR TITLE
App Layout

### LINKED ISSUE #8: Countries List
## Background
Create the countries list, so the user can view all available countries 

### UAT
- [ ] Create a Card component
  - [ ] card component accepts (image URL, name, population, region, capital ) as props 
  - [ ] Make sure the design of the card matches the UI provided in the frontend mentor website
- [ ] Create a countries list component
  - [ ] convert the countries list to an array of objects
  - [ ] loop through countries list
  - [ ] use the Card component to render each country
  
  
image

### PR DESCRIPTION
**Pull request by ireemalqarni**
Closes #8 and #9 

-Description
Add a new file to the application

-Change log
New Component added named Card.jsx contains a list of countries shows in a card with 2 features ( filter & search)

-Proof of success

https://github.com/user-attachments/assets/4ac683aa-6952-4e71-9119-5601d1eadaa9

How to test
- npm run  dev

### CODE REVIEW

**Review by MaymoonaAlBoloshi — CHANGES_REQUESTED** [review]
Please rebase your code to the main/master branch

File: src/App.tsx [hunk 1]
```diff
@@ -1,35 +1,15 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
 import './App.css'
+import Header from './layout/Header.jsx';
```
Comment (MaymoonaAlBoloshi) [inline src/App.tsx:2]:
```suggestion
import Header from './layout/Header';
```
remove the extension of the file on imports, also for the line bellow this

File: src/App.tsx [hunk 2]
```diff
@@ -1,35 +1,15 @@
...
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
+      <Header/>
```
Comment (MaymoonaAlBoloshi) [inline src/App.tsx:8]: import header in the main layout, and import main layout here

File: src/layout/MainLayout.tsx
Comment (MaymoonaAlBoloshi) [inline src/layout/MainLayout.tsx:1]: Complete the layout component

**Review by MaymoonaAlBoloshi — CHANGES_REQUESTED** [review]

**Comment by ireemalqarni** [comment]
Closes #8  and #9 

-Description
Add a new file to the application

-Change log
New Component added named Card.jsx contains a list of countries shows in a card with 2 features ( filter & search)

-Proof of success

https://github.com/user-attachments/assets/a942bde5-5083-4696-a09d-296d2ffde5bc




- How to test
npm run dev

**Review by MaymoonaAlBoloshi — CHANGES_REQUESTED** [review]
looking good but here are some things to update
- Seperate card component as in the issue description
- make sure when we search a single country the cards look decent
- make sure that the filter and search stay on place if we search, right now if you search for a country they move down
- country card should match the design
image
- filter and search inputs should match the design
image
- make sure overall spacing is close enough

**Review by MaymoonaAlBoloshi — APPROVED** [review]
🔥 🚀  great job

**Pull request merged**